src/profile/v100/entity/account/appearance_id.rs

```diff
@@ -32,9 +32,7 @@ impl AppearanceID {
     }
 
     pub fn from_number_of_accounts_on_network(n: usize) -> Self {
-        Self {
-            value: (n % (Self::MAX as usize)) as u8,
-        }
+        Self::new((n % ((Self::MAX + 1) as usize)) as u8).unwrap()
```

Comment:
`+1` needed, to fix the bug! otherwise `Self::MAX % Self::MAX == 0`, but we want it to be  `Self::MAX % Self::MAX == Self::MAX`.